src/cli/utils/utils.go

```diff
@@ -64,3 +66,30 @@ func AddCommand(parent, c *cobra.Command) (*cobra.Command, *pflag.FlagSet) {
 
 	return c, c.Flags()
 }
+
+// separates the provided folders into two sets: folders that use a pathContains
+// comparison (the default), and folders that use a pathPrefix comparison.
+// Any element beginning with a path.PathSeparator (ie: '/') is moved to the prefix
+// comparison set.  If folders is nil, returns only containsFolders with any any matcher.
+func splitFoldersIntoContainsAndPrefix(folders []string) ([]string, []string) {
+	var (
+		containsFolders = []string{}
+		prefixFolders   = []string{}
+	)
+
+	if len(folders) == 0 {
+		return selectors.Any(), nil
+	}
+
+	// separate folder selection inputs by behavior.
+	// any input beginning with a '/' character acts as a prefix match.
+	for _, f := range folders {
+		if len(f) > 0 && f[0] == path.PathSeparator {
```

Comment:
I suppose none of our users will intend to select on an empty string.  Mostly likely case is that this would happen due to a scripting failure, yeah?  `--foo-type a,,c`.  Skipping that value here is sensible, but also hides the issue from the user (potentially with unintended consequences).  I think I'll put this suggestion through, but also raise a ticket for pre-validation on these inputs as a separate step.